### PR TITLE
refactor(BreadcrumbItem): MenuItem support for customized key

### LINKED ISSUE #41432: Breadcrumb items配置menus后，menu的onClick事件获取的key是数组的index不是menu item的实际key
### Steps to reproduce

```javascript

  const options = [{id: 'id1', name: 'name1'}, {id: 'id2', name: 'name2'}, ].map((data) => {
    return {
      key: data.id,
      label: data.name,
    };
  });


<Breadcrumb.Item
          key="breadcrumb"
          menu={{
            items: options,
            onClick: (e) => {
              console.log(e.key)
            },
          }}
        >
          
        </Breadcrumb.Item>


```

### What is expected?

console.log(key)  key是 id1, id2

### What is actually happening?

console.log(key)  key是 1, 2

| Environment | Info |
| --- | --- |
| antd | 5.3.2 |
| React | ^18.2.0 |
| System | windows |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/41432
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  `MenuItem` support for customized key          |
| 🇨🇳 Chinese |  `MenuItem` 支持自定义 key          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 1]
```diff
@@ -80,7 +81,7 @@ const BreadcrumbItem: CompoundedComponent = (props: BreadcrumbItemProps) => {
 
             return {
               ...itemProps,
-              key: index,
+              key: key || index,
```
Comment (li-jia-nan) [inline components/breadcrumb/BreadcrumbItem.tsx:84]:
```suggestion
              key: key ?? index,
```

Comment (yoyo837) [inline components/breadcrumb/BreadcrumbItem.tsx:84]: 这里应该改成始终不用index, 没key开发者自己看警告吧

Comment (zombieJ) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 兜底其实还行

Comment (zombieJ) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 搞个测试用例？

Comment (li-jia-nan) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 没有传 key 或者存在重复 key 的时候给个报错吧

Comment (yoyo837) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 合并呢, react本身就会警告啊, 搞复杂了吧

Comment (yoyo837) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 兜底key我个人都不想要 
image

Comment (Yuiai01) [reply components/breadcrumb/BreadcrumbItem.tsx:84]: 来了来了，小弟这边修改一下加上～

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
好像 CI 有点问题。

**Comment by li-jia-nan** [comment]
我来修吧

**Comment by li-jia-nan** [comment]
> 好像 CI 有点问题。

- #41451

**Pull request merged**